jmclient/jmclient/client_protocol.py

```diff
@@ -283,52 +293,48 @@ def unconfirm_callback(self, txd, txid):
             if v["txd"]["outs"] == txd["outs"]:
                 offerinfo = v
                 break
-        if not offerinfo:
-            jlog.info("Failed to find notified unconfirmed transaction: " + txid)
-            return
-        removed_utxos = self.client.wallet.remove_old_utxos(txd)
-        jlog.info('saw tx on network, removed_utxos=\n{}'.format('\n'.join(
-            '{} - {}'.format(u, fmt_tx_data(tx_data, self.client.wallet))
-            for u, tx_data in removed_utxos.items())))
+        if offerinfo is None:
```

Comment:
Logging this case would be beneficial to aid debugging. The line could be simplified to `else:` instead of the `if` check.

Reply:
Not really, I think.
The rather big redesign here is that we can call the unconfirm callback lots of times on transactions which are not the one we're tracking (mostly that won't actually happen, but it can, and not only for joins in parallel); if it's not the one we're tracking, we return False so we know to keep looking for it, and only return True when we get a match. I don't think it's useful to log non-matches as there can be tons of them and they don't tell us much (except in a development phase).

The point about `else` is right though (I should get used to that!), 'fixed' in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/359/commits/e645104391bd07290d31e4ee9f6b6cd3f32c74a4

Reply:
I see, I was of the impression this is mostly an error condition. Maybe still and a `jlog.debug` just in case?

Reply:
I mean you could do it for testing, but I don't want to log every time a callback gets fired on an inapplicable transaction.